experiment/measurer/measure_worker.py

```diff
@@ -98,18 +97,23 @@ def _get_previous_cycle_state(self, cycle_dependent=True) -> list:
 
         return json.loads(result.output)
 
-    def get_previous(self, cycle_dependent=True):
+    def get_previous(self):
         """Returns the previous state."""
         if self._prev_state is None:
-            self._prev_state = self._get_previous_cycle_state(cycle_dependent)
+            self._prev_state = self._get_previous_cycle_state()
 
         return self._prev_state
 
-    def set_current(self, state, cycle_dependent=True):
+    def set_current(self, state, delete_previous_state=False):
         """Sets the state for this cycle in the bucket."""
         state_file_bucket_path = self._get_bucket_cycle_state_file_path(
-            self.cycle if cycle_dependent else 0)
+            self.cycle)
         with tempfile.NamedTemporaryFile(mode='w') as temp_file:
             temp_file.write(json.dumps(state))
             temp_file.flush()
             filestore_utils.cp(temp_file.name, state_file_bucket_path)
+        if delete_previous_state:
+            # Delete the state file for (current_cycle - 2)th cycle.
+            old_state_file_bucket_path = self._get_bucket_cycle_state_file_path(
+                self.cycle - 2)
```

Comment:
You don't need to delete if self.cycle <=2.

Reply:
Changed it.